hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -278,6 +281,11 @@ public ProxyService getProxyService() {
 
     @Override
     public Address getThisAddress() {
+        if (advancedNetworkConfigEnabled) {
+            Address clientServerSocketAddress = node.getLocalMember().getAddressMap().get(CLIENT);
+            assert clientServerSocketAddress != null;
```

Comment:
Probably I'm getting something wrong, but I'm unable to find the code that guarantees `clientServerSocketAddress != null`.

Reply:
`ClientEngineImpl` is only instantiated when a client server socket is configured, see https://github.com/hazelcast/hazelcast/blob/92ca85687ad1f9cc2057a7cd3b7a27f22bad8019/hazelcast/src/main/java/com/hazelcast/instance/Node.java#L249

Reply:
Thanks! Now I feel much better :)